### PR TITLE
Resolved paths in generated C-file's sources list

### PR DESCRIPTION
**Pull request by TSonono**
I did not write any tests for this since I'm not familiar with the repo. I tested it by "manually at least". Let me know if I should change something

### CODE REVIEW

**Review by MagnusIntel — APPROVED** [review]

File: py/dml/c_backend.py [hunk 1]
```diff
@@ -3247,7 +3247,7 @@ def generate_cfile(device, footers,
         ' * Generated by dmlc, do not edit!',
         ' *',
         ' * Source files:',
-        '\n'.join([' *   ' + f for f in source_files]),
+        '\n'.join([' *   ' + str(Path(f).resolve()) for f in source_files]),
```
[the remark this answers is not in the document]
Comment (TSonono) [reply py/dml/c_backend.py]: Thanks for the suggestion, fixed

**Review by lwaern-intel — CHANGES_REQUESTED** [review]
As stated in R.C., the `dead_dml_methods` script should be changed to assert on that the path is absolute. Something like:
```python
def dml_sources(c_file: Path) -> set[Path]:
    '''Given a DMLC-generated C file, return the set of DML files that
    were used to generate it'''
    def paths():
        for p in dml_sources_from_body(c_file.read_text()):
            p = Path(p)
            assert p.is_absolute(), p
            yield p
    return {p for p in paths()}
```

You're welcome to shorten that. Building a set by leveraging `.add()` is OK, I just wrote a generator adjustor for principle's sake.

Also, to test that the change and assertion works, modify the import in `tests/1.4/misc/dead_methods.dml` to
`import "../misc/dead_methods_imported.dml";`, with a comment stating why the import's like that (test that DMLC resolves the path in the source lists).

File: py/dead_dml_methods.py [hunk 2]
```diff
@@ -56,10 +56,14 @@ def dml_sources_from_body(body):
...
 def dml_sources(c_file: Path) -> set[Path]:
-    '''Given a DMLC-generated C file, return the set of DML files that
-    were used to generate it'''
-    return {(c_file.parent / Path(p)).resolve()
-            for p in dml_sources_from_body(c_file.read_text())}
+    """Given a DMLC-generated C file, return the set of DML files that
+    were used to generate it"""
+    def paths():
+        for p in dml_sources_from_body(c_file.read_text()):
+            p = Path(p)
```
Comment (lwaern-intel) [inline py/dead_dml_methods.py:63]: Cleaner:
```python
        for p in map(Path, dml_sources_from_body(c_file.read_text())):
            assert p.is_absolute(), f'{p} is not an absolute path'
            yield p
```

(thanks Erik!)

**Review by lwaern-intel — CHANGES_REQUESTED** [review]

File: py/dead_dml_methods.py (hunk 2 above)
Comment (TSonono) [reply py/dead_dml_methods.py:63]: Nice

**Comment by syssimics** [comment]
Verification [#12619624](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=12619624): pass

**Review by lwaern-intel — APPROVED** [review]
👍

**Pull request merged**